### ISSUE TITLE
So that more users can read all the content on your site, improve color contrast between text and backgrounds

### ISSUE THREAD

**Issue by jenniferthibault** [NYCPlanning/labs-planninglabs.nyc-archive#5]
Hi, I'm Jen, a New York-based experience designer at 18F, and I am SO STOKED to see what you've done! So much open source design excitement. 

## Expected Behavior
The color contrast between text and background colors meet or exceed a ratio of [4.5:1 ](https://www.w3.org/WAI/WCAG20/quickref/#visual-audio-contrast-contrast). This threshold makes it more likely that folks with vision impairments or colorblindness will be able to distinguish text from background on your site.

## Actual Behavior
The orange and white color combinations do not meet the 4.5:1 contrast ratio, so users with low vision or colorblindness might have a hard time reading your content. 

- Your darkest orange and white measure 3.45:1 (important to note this applies to orange text on a white background, and white text on an orange background—the ratio is the same measurement for both)
- Your yellow and orange measure 2.5:1

This mostly impacts your heading styles, tags, and link styles:

<img width="1088" alt="screen shot 2017-06-12 at 12 01 19 pm" src="https://user-images.githubusercontent.com/11636908/27043083-f9f6f250-4f66-11e7-9750-97bcd04d2783.png">
<img width="624" alt="screen shot 2017-06-12 at 12 03 23 pm" src="https://user-images.githubusercontent.com/11636908/27043169-3a9106c0-4f67-11e7-89ce-9d23a0031838.png">

## This is issue is done when:
- [ ] The color contrast ratio of text and backgrounds on the site meets a ratio of 4.5:1

**Comment by jenniferthibault** [comment]
This issue is one that we struggled with in the 18F color palette ourselves, so I wanted to share our approach and some of what we learned in case it's helpful to you. Sorry for the one-two post/comment flood, but I also wanted to separate the issue itself from the resources below in case you decide to approach it differently.

We hit this with our `$.bright` blue against white, which is a high-energy, positive color combination that felt central to our brand, but didn't meet the 4.5:1 ratio. We ended up creating a new color (our `$.medium` that would let us use blue & white for text, and moved the bright color to non-text accents against white, or only as text in combination with our `.$dark` blue backgrounds. 

I file this knowing that oranges, yellows, and whites are VERY tricky to work with in accessible contrast ratios in a way that won't totally muddy up the tone, look, and feel of your site. 😬 

## Tools and resources that helped us build an accessible color palette

- An [Accessible color combination tool for your color palette](https://toolness.github.io/accessible-color-matrix/) — (disclosure: built in part by several folks who work at 18F as a side project) This will help you see which colors are safe to use for backgrounds and text combinations if you're aiming for the 4.5:1 ratio.
  - Here's [your current palette](https://toolness.github.io/accessible-color-matrix/?n=white&n=dutch%20white&n=still%20de%20grain%20yellow&n=mango%20tango&n=cocoa%20brown&n=black&v=FFFFFF&v=F0DCB8&v=FFC861&v=F18238&v=D96B27&v=000000) 
- [0to255](http://www.0to255.com/d96b27) : A tool to easily find lighter/darker colors related to your primary color
- [WebAIM's color contrast checker](http://webaim.org/resources/contrastchecker/) to measure the value of one specific combination

**For even more background on color accessibility:**
- [Color Accessibility Workflows](https://alistapart.com/article/color-accessibility-workflows), by Geri Coady. Posted last week on A List Apart
- The [18F Accessibility guide](https://accessibility.18f.gov/color/)

Glad to help think through this with you all to whatever extent you're interested, and again, awesome work!

**Comment by chriswhong** [comment]
Thank you so much for this feedback @jenniferthibault!  We will experiment with other color choices and try to achieve the 4.5:1 contrast ratio.  I'll post our trials here.

**Comment by andycochran** [comment]
Thanks, @jenniferthibault. This is definitely something we're considering. I'm working on revising the site design. Luckily we can easily meet the 3:1 contrast ratio required for AA compliance with large text. 

I propose that we make it a rule to never put small text on the orange background. And also darken the color of links to meet AA reqs for small text. 

![image](https://user-images.githubusercontent.com/409279/28633708-bad71136-7202-11e7-9e44-bf5da3a3327e.png)

^ These colors meet AA reqs for large text:
- [Headline: #ffffff on #d96b27](http://webaim.org/resources/contrastchecker/?fcolor=ffffff&bcolor=d96b27)
- [Button: #ffc861 on #ae561f](http://webaim.org/resources/contrastchecker/?fcolor=ffc861&bcolor=ae561f)

![image](https://user-images.githubusercontent.com/409279/28634051-f177d3be-7203-11e7-8f30-2cba812bff75.png)

^ And this is the least we can darken links color to meet AA reqs for small text:
- [Links: #ba5c21](http://webaim.org/resources/contrastchecker/?fcolor=ba5c21&bcolor=fff)

**Comment by andycochran** [comment]
And to use large yellow on orange it'll need to lighten up to [#ffeec1](http://webaim.org/resources/contrastchecker/?fcolor=ffeec1&bcolor=d96b27), which still has the effect of subduing the text to be less prominent than the white text. 

![image](https://user-images.githubusercontent.com/409279/28634697-070d202e-7206-11e7-89e0-395d39b02a2e.png)

**Comment by andycochran** [comment]
Actually, `#AE561F` and `#BA5C21` are so close that I'll just stick with the one that's AA-compliant for small text.

**Comment by andycochran** [comment]
This has been addressed in [our new site repo](https://github.com/NYCPlanning/labs-planninglabs.nyc/). Much thanks to @jenniferthibault for this thoughtfully- and thoroughly-crafted issue.
